sql/expression/convert.go

```diff
@@ -240,3 +246,18 @@ func convertValue(val interface{}, castTo string, originType sql.Type) (interfac
 		return nil, nil
 	}
 }
+
+// convertHexBlobToDecimalForNumericContext returns numeric value if the value is binary and origin type is blob
+// IF the convertTo type is number type. This applies only for hex LITERAL values parsed it is parsed as
+// BLOB type in the parser but to binary string type.
+// -- other byte arrays of other sql types does not apply here.
```

Comment:
Is `--` necessary in the comments here?
And "does not apply here" doesn't accurately convey what happens.

Maybe something like this?
```
// Byte arrays of other SQL types are not handled here.
```